scipy/stats/_continuous_distns.py

```diff
@@ -6376,10 +6376,87 @@ def _stats(self):
     def _entropy(self):
         return _EULER/2.0 + 1 - 0.5*np.log(2)
 
+    def fit(self, data, *args, **kwds):
+        """
+        Maxmimum likelihood estimation of the `scale` parameter
+        is used in the numerical optimization to determine `loc`.
+
+        Parameters
+        ----------
+        data : array_like
+            Data to use in calculating the maximum likelihood estimate.
+
+        Returns
+        -------
+        loc, scale : float
+            Maximum likelihood estimate for the scale, provided loc.
+
+        Notes:
+        If `floc` is fixed, the maximum likelihood scale may be calculated
+        without numerical optimization. Otherwise, numerical optimization
+        from `.rv_continuous` will be  conducted to determine `loc` and its
+        scale based on MLE.
+        """
+
+        data, floc, fscale = _check_fit_input_parameters(self, data,
+                                                         args, kwds)
+
+        def scale_mle(loc, data):
+            # Source: Statistical Distributions, 3rd Edition. Evans, Hastings,
+            # and Peacock (2000), Page 175
+            return (np.sum((data - loc) ** 2) / (2 * len(data))) ** .5
+
+        """
+        if `floc` is not fixed, provide a `_reduce_func` for `.rv_continuous`
+        to use. '_reduce_func' returns:
+            - the parameter to optimize from `x0`, loc
+            - `func` to minimize, and
+            - `get_tuple` to obtain the dependent parameter, as
+               numerical optimization over `func` only indicates
+               the independent parameter's solution.
+        """
+
+        if floc is not None:
+            return floc, scale_mle(floc, data)
+        if fscale is not None:
+            # `fscale` is fixed, use standard numerical optimization
+            return super(rayleigh_gen,
+                         self).fit(data, *args, **kwds)
+        else:
+            # account for user provided guesses
+            loc, scale = self._fitstart(data)
+            loc = kwds.pop('loc', loc)
+            scale = kwds.pop('scale', scale)
+
+            x0, ll, restore, args = self._reduce_func((loc, scale), kwds)
+
+            opt = _fit_determine_opt(kwds.pop('optimizer', optimize.fmin))
+
+            # wrap LL to optimize over `loc`
+            def func(loc, data):
+                return ll([loc, scale_mle(loc, data)], data)
+
+            loc = opt(func, x0[0], args=(data,), disp=0)
+
+            return loc[0], scale_mle(loc, data)
+
 
 rayleigh = rayleigh_gen(a=0.0, name="rayleigh")
 
 
+def _fit_determine_opt(optimizer):
```

Comment:
This is probably one of the reasons we wanted to call the superclass `fit` method - to avoid repeating this. Not required, but what do you think about breaking this out of the superclass `fit` method as a function so you can import it here, too?   Even though it requires modifying the superclass `fit` method, I don't think there's much to argue about it since it's just a refactoring; it doesn't change how it can be used or overridden.

Reply:
I think that could be appropriate.